x-pack/plugin/deprecation/qa/rest/src/main/java/org/elasticsearch/xpack/deprecation/TestDeprecationHeaderRestAction.java

```diff
@@ -90,12 +92,19 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
             }
         }
 
+        // Pull out the settings values here in order to guarantee that any deprecation messages are triggered
+        // in the same thread context.
+        final Map<String, Object> settingsMap = new HashMap<>();
```

Comment:
For some reason, I was tripping over an issue where deprecations in the thread context seemed to get lost during processing. In the end I attributed it to #19222 and reworked the code.